### ISSUE TITLE
Fedora can't build Mono project

### ISSUE THREAD

**Issue by Whimfoome** [godotengine/godot#56155]
### Godot version

4.0.dev (489f11ee0cbbb90f4f629aa7fd846078148aaa8e)

### System information

Fedora Linux 35, AMDGPU, Wayland

### Issue description

Can't build Mono project in Godot 4 on Fedora.

I have tried Arch based distros and Ubuntu based distros and they worked with the Mono and Dotnet packages installed.
Now I am testing Fedora and installed mono and dotnet.
When trying to build a project I get this error:
![image](https://user-images.githubusercontent.com/42871796/147070569-c28526b6-ddef-40f5-bed3-01e28c172de9.png)

It's probably because it's trying to build with Mono, but I want to build with dotnet cli. The problem is the options are missing on Fedora:
![image](https://user-images.githubusercontent.com/42871796/147070837-495ee16d-d298-42ce-ac9b-4e9b83cd3f2a.png)

This is happening only in Godot 4. Under Godot 3.4, the options are there
![image](https://user-images.githubusercontent.com/42871796/147071568-a421d592-4d71-4c10-be53-4b34fc329b1e.png)
![image](https://user-images.githubusercontent.com/42871796/147071594-90ff4bd7-0fec-419e-b7a0-01ec5954ade4.png)

This is only under Fedora so far. With Ubuntu based distros and Arch based distros I get the options in Godot 4, just not in Fedora.
![image](https://user-images.githubusercontent.com/42871796/147071744-da22471e-be79-4fde-9e1e-e1ff9886ef9a.png)
![image](https://user-images.githubusercontent.com/42871796/147071783-8b2a4cac-e1d2-463c-8fb0-754fa0a02667.png)

### Steps to reproduce

1. Install Fedora based distro - Fedora Workstation or Silverblue
2. Install package `dotnet-sdk-5.0.x86_64` and additionally dotnet 6, until it's supported officially
3. Install package `mono-devel`
4. Try to build Mono project with Godot 4 on Fedora and on other distros.
5. The options in Fedora are missing in Editor Settings

**Comment by raulsntos** [comment]
It seems something is wrong with your mono setup, it seems in Fedora the `msbuild` tool is a separate package so it also needs to be installed (see https://github.com/godotengine/godot/issues/55610#issuecomment-985987021).

Issue #52211 is similar and may contain additional helpful information, but as a summary make sure the `GodotSharp` folder is next to godot's binary and check that it contains the expected files with `find GodotSharp/ | sort` as described in https://github.com/godotengine/godot/issues/52211#issuecomment-919453816.

**Comment by Whimfoome** [comment]
When I type `msbuild` I get:
![image](https://user-images.githubusercontent.com/42871796/147121092-dd643b89-9fc8-4960-8a94-816de46ab132.png)
but it's working with `dotnet msbuild`:
![image](https://user-images.githubusercontent.com/42871796/147121154-aab86f92-ccc1-4c50-9b82-1ad3efabd580.png)

About the other concern, the folder is in place with everything in it. I basically download the compiled files from GitHub Actions that contain the Editor with Mono.

I get this problem only on Fedora (Workstation and Silverblue).

**Comment by raulsntos** [comment]
Ah I see, you are not building the editor yourself, in that case I don't think you need mono, dotnet should be enough. If you do want to use mono though, you'll need to install MSBuild since it's a separate package in Fedora (not included in `mono-devel` or `mono-complete`).

I also want to clarify, just in case, that the `msbuild` command is the mono version of MSBuild and the `dotnet msbuild` command is using Microsoft's .NET Sdk (the **dotnet CLI**) and these are two different tools. Godot requires the .NET Sdk but mono is optional (only required if you set the build tool to **MSBuild (mono)**), the default build tool should be set to **dotnet CLI**. However, it seems that Godot is unable to get to this point in your case since it doesn't even have the editor settings.

Can you run Godot from the terminal and see if it prints a full version of the `TypeInitializationException` there with the stack frames included? It would be useful to try and see where the exception originates from, also there may be other useful debugging information that is not printed to the Godot console.

**Comment by akien-mga** [comment]
For the record, the CI artifacts are not tested to be functional to compile C# scripts. They're Mono builds only to maximize code coverage on compilation, but the packaging might not be sufficient to have a working Mono distrib.

**Comment by Whimfoome** [comment]
>  don't think you need mono, dotnet should be enough

If I run Godot without mono installed I get `./godot.linuxbsd.opt.tools.64.mono: error while loading shared libraries: libmonosgen-2.0.so.1: cannot open shared object file: No such file or directory`

> Can you run Godot from the terminal and see if it prints a full version of the TypeInitializationException there with the stack frames included?

Sorry, but I returned to Arch-based distro as I needed to do some work. Here everything is working as expected.
Screenshot from Godot 4 ([this build](https://github.com/godotengine/godot/actions/runs/1612257028)) on EndeavourOS, dotnet-sdk-6 and mono installed:
![image](https://user-images.githubusercontent.com/42871796/147221584-8577963e-eb60-4217-9cb1-67ab62326d8f.png)

I would be happy to install Fedora on another partition and test this issue again if there are other people with this problem.
